src/core/compact_object.h

```diff
@@ -445,13 +448,21 @@ class CompactObj {
     };
   } __attribute__((packed));
 
+  struct JsonConsT {
+    JsonType* json_ptr;
+    size_t bytes_used;
```

Comment:
why not reduce it to uint32_t and keep the encoding inside the object?

Reply:
uint_32 limits to 4gb. It's extreme to expect any object would take more than that but you never know. As for enconding, I don't think it makes much sense to store this. It's fixed once it's set and that's also one of the reasons I refactored it

Reply:
The reason for storing it per object is that I expect we would want a heuristic at some point for encoding, managed per object basis.  like we do with hash maps, sets etc. flat is awful for mutable operations, for example.  we do not have to do it now though.

Reply:
I wanted to pack this in the beginning and in fact I already mentioned to @adiholden that if in the future we wanted more space on the json we store within a compcat object we could get some. However I thought that the size_t suffices for now and until we actually need this there is no reason to take the bitfield route.

Reply:
also if we ever want a heuristic, how many bits would that require? 4-8 ? For that use case we will just make the bytes_used (and the length for flat json) a bitfield and mask the last `n` bits we want to use for enconding. It would be a simple change that would server our need.